Source/carthage/BuildVersion.swift

```diff
@@ -0,0 +1,56 @@
+//
+//  BuildVersion.swift
+//  Carthage
+//
+//  Created by Rodrigo Garcia on 10/15/16.
+//  Copyright © 2016 Carthage. All rights reserved.
+//
+
+import Foundation
+import CarthageKit
+import ReactiveCocoa
+import ReactiveTask
+import Result
+import Tentacle
+
+public struct CarthageVersion {
+	
+	public let localVersion: SemanticVersion
+	public let remoteVersion: SemanticVersion
+	
+	public init(localVersion: SemanticVersion, remoteVersion: SemanticVersion) {
+		self.localVersion = localVersion
+		self.remoteVersion = remoteVersion
+	}
+	
+	func shouldUpgrade() -> Bool {
+		return localVersion < remoteVersion
+	}
+}
```

Comment:
I don't think this struct gets us anything. Lets get rid of it and just use `localVersion` and `remoteVersion` directly.